spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubChannelAdaptersIntegrationTests.java

```diff
@@ -80,103 +72,120 @@
  *
  * @author João André Martins
  * @author Mike Eltsufin
+ * @author Elena Felder
  */
 public class PubSubChannelAdaptersIntegrationTests {
 
 	private static final int RECEIVE_TIMEOUT_MS = 10000;
 
-	private static PrintStream systemOut;
+	static PubSubAdmin pubSubAdmin;
 
-	private static ByteArrayOutputStream baos;
+	/** Capture output for verification. */
+	@Rule
+	public OutputCaptureRule outputCaptureRule = new OutputCaptureRule();
 
-	private ApplicationContextRunner contextRunner = new ApplicationContextRunner()
-			.withConfiguration(AutoConfigurations.of(
-					GcpContextAutoConfiguration.class,
-					GcpPubSubAutoConfiguration.class))
-			.withUserConfiguration(PollableConfiguration.class, CommonConfiguration.class);
+	String topicName;
+
+	String subscriptionName;
+
+	ApplicationContextRunner contextRunner;
 
 	@BeforeClass
 	public static void enableTests() {
 		assumeThat(System.getProperty("it.pubsub")).isEqualTo("true");
 	}
 
 	@BeforeClass
-	public static void captureStdout() {
-		systemOut = System.out;
-		baos = new ByteArrayOutputStream();
-		TeeOutputStream out = new TeeOutputStream(systemOut, baos);
-		System.setOut(new PrintStream(out));
+	public static void initializePubSubAdmin() throws Exception {
+		pubSubAdmin = new PubSubAdmin(
+				new DefaultGcpProjectIdProvider(),
+				new DefaultCredentialsProvider(() -> new Credentials())
+		);
 	}
 
-	@AfterClass
-	public static void resetStdout() {
-		System.setOut(systemOut);
+	@Before
```

Comment:
Do we want to do this per test, or once for the whole class?

Reply:
I would at least create a subscription per test; otherwise each test has to take care to never have anything redelivered.

At that point, might as well create a fresh topic.

Reply:
fair enough... feel free to merge.